src/app/frontend/replicasetdetail/replicaset_service.js

```diff
@@ -64,4 +66,41 @@ export class DeleteReplicaSetService {
 
     return deferred.promise;
   }
+
+  /**
+   * Opens an update replica set dialog. Returns a promise that is resolved/rejected when user wants
+   * to update the replicas. Nothing happens when user clicks cancel on the dialog.
+   *
+   * @param {!backendApi.ReplicaSetDetail} replicaSetDetail
+   * @returns {!angular.$q.Promise}
+   */
+  showUpdateReplicasDialog(replicaSetDetail) {
+    return showUpdateReplicasDialog(
+        this.mdDialog_, replicaSetDetail, this.updateReplicas.bind(this));
+  }
+
+  /**
+   * Updates number of replicas in replica set.
+   *
+   * @param {number} replicasCount
+   * @param {!backendApi.ReplicaSetDetail} replicaSetDetail
+   * @param {!function(!backendApi.ReplicaSetSpec)=} opt_callback
+   * @param {!function(!angular.$http.Response)=} opt_errback
+   * @return {angular.$q.Promise}
+   */
+  updateReplicas(replicasCount, replicaSetDetail, opt_callback, opt_errback) {
```

Comment:
Hmm... I think that you can move this method to UpdateReplicasDialog. With this we would get rid of the lengthy callback. You can just make showUpdateReplicasDialog return the promise of this function. The same as delete dialog. 
What do you think?